src/Units/KaijuActions.java

```diff
@@ -0,0 +1,943 @@
+package Units;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import Engine.GameAction;
+import Engine.GameActionSet;
+import Engine.GamePath;
+import Engine.UnitActionFactory;
+import Engine.Utils;
+import Engine.XYCoord;
+import Engine.Combat.DamagePopup;
+import Engine.GameEvents.ArmyDefeatEvent;
+import Engine.GameEvents.GameEventQueue;
+import Engine.GameEvents.HealUnitEvent;
+import Engine.GameEvents.MapChangeEvent;
+import Engine.GameEvents.MassDamageEvent;
+import Engine.GamePath.PathNode;
+import Engine.StateTrackers.StateTracker;
+import Engine.UnitActionLifecycles.TransformLifecycle;
+import Terrain.Environment;
+import Terrain.GameMap;
+import Terrain.MapLocation;
+import Terrain.MapMaster;
+import Terrain.TerrainType;
+import Units.KaijuWarsKaiju.HellTurkey;
+import Units.KaijuWarsKaiju.HellTurkeyLand;
+import Units.KaijuWarsKaiju.KaijuStateTracker;
+import Units.KaijuWarsKaiju.KaijuUnitModel;
+import Units.KaijuWarsUnits.KaijuWarsUnitModel;
+import Units.KaijuWarsUnits.Radar;
+
+public class KaijuActions
+{
+  public static class KaijuCrushFactory extends UnitActionFactory
+  {
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public GameActionSet getPossibleActions(GameMap map, GamePath movePath, Unit actor, boolean ignoreResident)
+    {
+      return new GameActionSet(new KaijuCrushAction(actor, movePath, this), false);
+    }
+
+    @Override
+    public String name(Unit actor)
+    {
+      return "CRUSH";
+    }
+  } //~Factory
+
+  /** List of things Kaiju break that aren't capturable */
+  public static final List<TerrainType> INERT_CRUSHABLES = Arrays.asList(TerrainType.BRIDGE, TerrainType.BUNKER, TerrainType.PILLAR, TerrainType.METEOR);
+  private static boolean isCrushable(MapLocation location)
+  {
+    return location.isCaptureable()
+        || INERT_CRUSHABLES.contains(location.getEnvironment().terrainType);
+  }
+
+  public static class KaijuCrushAction extends GameAction
+  {
+    private final KaijuCrushFactory type;
+    private final GamePath movePath;
+    private final XYCoord waitLoc;
+    private final Unit actor;
+
+    public KaijuCrushAction(Unit unit, GamePath path, KaijuCrushFactory pType)
+    {
+      type = pType;
```

Comment:
Why does the constructor need to accept the `KaijuCrushFactory` parameter? Is it just so we have the object to return from `getType()`? For that matter, why do all the other game actions just return `null` from getType()`?

Reply:
Yeah, it's just to have something to return for `getType()`.
`getType()` is used by AIs (for organizing) and by the game input machinery (for control flow and its `toString()`)

`GameAction`s that aren't taken by a unit (i.e. put into a `GameActionSet`) don't need one because they aren't constructed via `UnitActionFactory`.
`GameAction` types defined in a lifecycle file have a real `getType()` because they're produced in groups, and the caller has to have some way of categorizing/interrogating them.

I had considered renaming it `getActionType()`, but that provides no extra clarity.
`getActionFactory()` could work, I guess?

Reply:
Gotcha. Is there an advantage to passing it in here instead of referencing the type statically like we do elsewhere? Just curious why it's handled differently here.

Reply:
`KaijuAttackAction` is a generic parent for basically everything in here, so it doesn't know which class to point to.
So I'm left choosing between:
Make each action type implement that static linkage (function override -> 4 lines)
`KaijuAttackAction` handles it, and the type object is passed in (add params to a bunch of constructors -> +~7 characters on a bunch of lines)

I defaulted to the second, I think, since I was copying code from `TransformLifecycle` (which needs the specific type object so it actually shows the user the right thing).

It might also be possible to elide a few concrete `Action` classes entirely (and make `KaijuAttackAction` not abstract). I'd have to look a bit closer at it.